modules/doc/content/training/darcy_thermo_mech/step01_moose_object.md

```diff
@@ -0,0 +1,124 @@
+# Step 1: Creating and Executing a MOOSE Object
```

Comment:
The MooseObject stuff should come in step 2, before the first Kernel is created. This step should focus on Diffusion and the input file.

Reply:
The reason I included that first was because I though it related well to input file syntax. Perhaps I should only briefly mention them and then move this more in-depth discussion to step 2. Although, I wanted to talk about Galerkin FEM and weak forms in step 2, so that step would become cluttered. Let me know what you think of this.

Reply:
Step 2 can just be MooseObject; we can inherit from Diffusion originally so can leave all the FEM stuff for step3, which can use KernelGrad as a base class. This is just an idea, what do you think?

Reply:
I like this idea, we should spread out concept discussions as much as possible. 

I'm thinking I subdivide FEM even further, and only talk about weighted residuals and generating a weak form in step 3. Then, maybe step 4 can be about polynomial fitting, test functions = shape functions in Galerkin FEM, jacobians, etc. Finally step 5 can be about integration, i.e., Gauss quad rules. If I have to create all new tutorial steps to make this work I will.

Anyways, first I'll work on step 2 with MOOSE objects, then we'll worry about FEM steps.